File: app/src/main/java/com/kongpf8848/dmail/login/oauth/yahoo/YahooApi.java
package com.kongpf8848.dmail.login.oauth.yahoo;

import com.kongpf8848.dmail.login.oauth.AuthorizationHeader;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Header;

public interface YahooApi {

    @GET("/openid/v1/userinfo")
    Single<YahooProfileResponse> profile(@Header("Authorization") AuthorizationHeader header);
}

File: app/src/main/java/com/kongpf8848/dmail/login/oauth/yahoo/YahooApi.kt
package com.kongpf8848.dmail.login.oauth.yahoo

import retrofit2.http.GET
import com.kongpf8848.dmail.login.oauth.AuthorizationHeader
import com.kongpf8848.dmail.login.oauth.yahoo.YahooProfileResponse
import io.reactivex.Single
import retrofit2.http.Header

interface YahooApi {
    @GET("/openid/v1/userinfo")
    fun profile(@Header("Authorization") header: AuthorizationHeader): Single<YahooProfileResponse>
}